internal/store/endpoint.go

```diff
@@ -150,6 +151,90 @@ func endpointMetricFamilies(allowAnnotationsList, allowLabelsList []string) []ge
 				}
 			}),
 		),
+		*generator.NewFamilyGenerator(
+			"kube_endpoint_address_available",
+			"Addresses available in endpoint",
+			metric.Gauge,
+			"",
+			wrapEndpointFunc(func(e *v1.Endpoints) *metric.Family {
+				var addresses string
+				var ports string
+				ms := []*metric.Metric{}
+				for _, s := range e.Subsets {
+					for _, address := range s.Addresses {
```

Comment:
To simplify the logic here, I would suggest building a new string slice with the `ip:port` for each endpoint and using `strings.join` at the end.